boards/arm/arduino_nano_33_ble/arduino_nano_33_ble.dts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2020 Jefferson Lee
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/dts-v1/;
+#include <nordic/nrf52840_qiaa.dtsi>
+
+/ {
+    model = "Arduino Nano 33 BLE (Sense)";
+    compatible = "arduino,arduino_nano_33_ble";
+
+    chosen {
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+		zephyr,uart-mcumgr = &uart0;
+		zephyr,bt-mon-uart = &uart0;
+		zephyr,bt-c2h-uart = &uart0;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,code-partition = &code_partition;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led0: led_0 {
+			gpios = <&gpio0 24 0>;
+			label = "Red LED";
+		};
+	};
+
+	/* These aliases are provided for compatibility with samples */
+	aliases {
+		led0 = &led0;
+		spi = &spi2;
+	};
+};
+
+&flash0 {
+	/*
+	 * For more information, see:
+	 * https://docs.zephyrproject.org/latest/guides/dts/legacy-macros.html#legacy-flash-partitions
+	 * these bindings are irrelevant. See the LD script
+	 */
+	partitions {
+		compatible = "fixed-partitions";
```

Comment:
So you're defining only the code-partition here, nothing about bootloader or storage (see other ARM boards for reference).

Since the code-partition does not start at 0x0, maybe you could define the area from 0x0 to 0x10000 as the bootloader partition? Otherwise, how would the image, loaded in the code-partition be booted?

I am also wondering what happens to the flash area above the code partition - why is it not included in the code-partition - is it used by sthg else?

Reply:
The Arduino has their own BOSSA-adjacent bootloader in the region from 0-0x10000. If I tag this region as bootloader do I need to do anything to make sure that MCUBoot doesn't get written over the Arduino-supplied bootloader (it's a pain to flash the original back).

Reply:
OK, I see that you are using a custom linker script with the following snippet:

```
MEMORY
{
  FLASH (rx) : ORIGIN = 0x10000, LENGTH = 0xf0000
  RAM_NVIC (rwx) : ORIGIN = 0x20000000, LENGTH = 0x100
  RAM_CRASH_DATA (rwx) : ORIGIN = (0x20000000 + 0x100), LENGTH = 0x100
  RAM (rwx) : ORIGIN = ((0x20000000 + 0x100) + 0x100), LENGTH = (0x40000 - (0x100 + 0x100))
}
```
This one seems to overwrite the common ARM linker script. So it basically overrides all the DeviceTree / Kconfig infrastructure for defining flash and ram areas.

Could you get rid of this, and find a way that you can use the common ARM linker script "as is"? At least for FLASH and SRAM.

For example, I see that you would like to have your code area starting from 0x10000 and with size 0xf0000. Should that **always** be the case? If so, you can define your code partition as follows:

```
code_partition: partition@10000 {
			label = "code";
			reg = <0x10000 0xf0000>;
			read-only;
};
```
And (as it is already the case in the .dts) choose it:

```
zephyr,flash = &flash0;
zephyr,code-partition = &code_partition;
```

For this to be enforced by the linker, you need to enable the CONFIG_USE_DT_CODE_PARTITION in your board definition.

This has nothing to do with MCUboot.
And, by the way, MCUboot requires a partition named "boot_partition", but you do not define such thing here. So, MCUboot won't work. If you do not need MCUboot, you can ignore.

Reply:
Hi @saltyJeff , you should follow what @ioannisg recommended. I complement with, you may need look at https://github.com/zephyrproject-rtos/zephyr/issues/29312 and https://github.com/zephyrproject-rtos/zephyr/pull/29874. to be aware. The have BOSSAC fully integrated with Zephyr west tool, you need SDK 0.12.0 (https://github.com/zephyrproject-rtos/sdk-ng/releases/tag/v0.12.0-beta-1) to get support to SAM-BA compatible bootloaders and 29874.

As @ioannisg said, you need CONFIG_USE_DT_CODE_PARTITION . However, if you use BOSSAC Arduino/Adafruit UF2 compatible you need define at \<board\>_defconfig
CONFIG_BOOTLOADER_BOSSA=y
CONFIG_BOOTLOADER_BOSSA_ARDUINO=y or CONFIG_BOOTLOADER_BOSSA_ADAFRUIT_UF2=y
This will auto select CONFIG_USE_DT_CODE_PARTITION and enables the board to reset and enter on bootloader mode too.

You can look at **arduino_nano_33_iot** or **adafruit_itsybitsy_m4_express** as reference.